File: 383.赎金信.py
#
# @lc app=leetcode.cn id=383 lang=python3
#
# [383] 赎金信
#

# @lc code=start
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        words=[0]*26
        for c in magazine:
            words[ord(c)-ord('a')]+=1
        for c in ransomNote:
            index=ord(c)-ord('a')
            words[index]-=1
            if words[index]<0:
                return False
        return True
# @lc code=end

# Solution1:
# class Solution:
#     def canConstruct(self, ransomNote: str, magazine: str) -> bool:
#         words=[0]*26
#         for c in magazine:
#             words[ord(c)-ord('a')]+=1
#         for c in ransomNote:
#             index=ord(c)-ord('a')
#             words[index]-=1
#             if words[index]<0:
#                 return False
#         return True

File: review/383.赎金信.java
/*
 * @lc app=leetcode.cn id=383 lang=java
 *
 * [383] 赎金信
 */

// @lc code=start
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] arr=new int[26];
        for(char c:magazine.toCharArray()){
            arr[c-'a']++;
        }
        for(char c:ransomNote.toCharArray()){
            arr[c-'a']--;
        }
        for(int n:arr){
            if(n<0){
                return false;
            }
        }
        return true;
    }
}
// @lc code=end

// Solution1:
// class Solution {
//     public boolean canConstruct(String ransomNote, String magazine) {
//         int[] arr=new int[26];
//         for(char c:magazine.toCharArray()){
//             arr[c-'a']++;
//         }
//         for(char c:ransomNote.toCharArray()){
//             arr[c-'a']--;
//         }
//         for(int n:arr){
//             if(n<0){
//                 return false;
//             }
//         }
//         return true;
//     }
// }